src/services/sessionMgr.js

```diff
@@ -617,6 +621,37 @@ const activateNextBlock = async ({ userId }) => {
     await cleanCache(shortname)
   }
 
+  // TODO: pubsub
+  const savedsession = user.runningSession.toObject()
+  let activeBlock
+  const arrayLength = savedsession.blocks.length
+  for (let i = 0; i < arrayLength; i += 1) {
+    if (savedsession.blocks[i].status === QUESTION_BLOCK_STATUS.ACTIVE) {
+      activeBlock = savedsession.blocks[i]
+    }
+  }
+
+  if (activeBlock != null) {
+    const instanceId = activeBlock.instances[0]
+    const questioninstance = await QuestionInstanceModel.findById(instanceId)
+    const question = await QuestionModel.findById(questioninstance.question)
+    const questionPublic = Object()
+    const questionInfo = question.versions[question.versions.length - 1]
+    questionPublic.id = instanceId
+    questionPublic.questionId = question._id
+    questionPublic.title = question.title
+    questionPublic.type = question.type
+    questionPublic.content = questionInfo.content
+    questionPublic.description = questionInfo.description
+    questionPublic.options = questionInfo.options
+    questionPublic.files = questionInfo.files
+    savedsession.activeInstances[0] = questionPublic
```

Comment:
Question: what happens when a block that is activated has multiple questions in it (i.e., multiple active instances?). As you are just doing your logic for the first element in the array in here.

Reply:
Would just need a loop that iterates over all instances that are newly active and adds these to the session, I think